src/index.spec.ts

```diff
@@ -191,6 +191,7 @@ describe('EppoJSClient E2E test', () => {
       apiKey,
       baseUrl,
       assignmentLogger: mockLogger,
+      forceReinitialize: true,
```

Comment:
A lot of this is pretty messy - ideally we would not need to do this in tests but the singletons persist across test cases. Maybe this trick is what's needed for an isolated test:

```
      jest.isolateModules(() => {
        // Isolate and re-require so that the static instance is reset to it's default state
        // eslint-disable-next-line @typescript-eslint/no-var-requires
        const reloadedModule = require('./index');
        init = reloadedModule.init;
        getInstance = reloadedModule.getInstance;
      });
```

Reply:
I think it's nice to be able to force re-initialize, for tests and possibly other applications